### PR TITLE
Remove unnecessary immediately invoked closures

### PR DESCRIPTION
**Pull request by souze**
For the tests, they where likely added before tests could return Result.
For the code, likely got easier to handle with thiserror crate

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -431,34 +431,28 @@ mod tests {
...
-            let should = crate::process::wait::WaitStatus::Signaled(
-                s.process.child_pid,
-                crate::process::signal::Signal::SIGTERM,
-                false,
-            );
-            assert_eq!(should, s.process.exit()?);
-            Ok(())
-        }()
-        .unwrap_or_else(|e| panic!("test_read_line failed: {}", e));
+    fn test_read_line() -> Result<(), Error> {
```
Comment (petreeftime) [inline src/session.rs:434]: nit: These tests don't need to return anything.

**Review by petreeftime — APPROVED** [review]

**Comment by petreeftime** [comment]
@matthiasbeyer I think bors is not enabled for this repo now.

**Comment by matthiasbeyer** [comment]
I requested it to be installed. :+1:

**Comment by petreeftime** [comment]
bors try

**Comment by petreeftime** [comment]
bors merge

**Pull request merged**